python/grass/script/setup.py

```diff
@@ -78,7 +78,7 @@
         print(vect)
 
     # clean up at the end
-    gsetup.cleanup()
+    gsetup.finish()
```

Comment:
Yes, it would! See #1834 for solution for notebooks (where `with` is not that useful). A similar approach can be used to implement a context manager to support usage with `with`.

For now, I documented the return value of `init` as something which can change (I think the original idea is that user uses that to delete the file because there was no `finish` and I will leave that for another PR or create an issue for it.

Reply:
The context manager (`with`) request moved to a new issue: #1848